libraries/azure-app-configuration-importer/src/appConfigurationImporter.ts

```diff
@@ -73,63 +134,35 @@ export class AppConfigurationImporter {
       srcKeyLabelLookUp[config.key][config.label || ""] = true;
     });
 
-    // generate correlationRequestId for operations in the same activity
-    const customCorrelationRequestId: string = uuidv4();
-    const customHeadersOption: OperationOptions = {
-      requestOptions: {
-        customHeaders: {
-          [Constants.CorrelationRequestIdHeader]: customCorrelationRequestId
-        }
+    configurationSettingToAdd.push(...configSettings);
+
+    for await (const existing of this.configurationClient.listConfigurationSettings({...configSettingsSource.FilterOptions, ...customHeadersOption})) {
+      const isKeyLabelPresent: boolean = srcKeyLabelLookUp[existing.key] && srcKeyLabelLookUp[existing.key][existing.label || ""];
+      if (strict && !isKeyLabelPresent) {
+        configurationSettingToDelete.push(existing);
       }
-    };
 
-    if (strict || importMode == ImportMode.IgnoreMatch) {
-      for await (const existing of this.configurationClient.listConfigurationSettings({...configSettingsSource.FilterOptions, ...customHeadersOption})) {
+      const incoming = configSettings.find(configSetting => configSetting.key == existing.key && configSetting.label === existing.label);
 
-        const isKeyLabelPresent: boolean = srcKeyLabelLookUp[existing.key] && srcKeyLabelLookUp[existing.key][existing.label || ""];
-        
-        if (strict && !isKeyLabelPresent) {
-          configurationSettingToDelete.push(existing);
-        }
-       
-        if (importMode == ImportMode.IgnoreMatch) {
-          const incoming = configSettings.find(configSetting => configSetting.key == existing.key && 
-            configSetting.label == existing.label);
-           
-          if (incoming && isConfigSettingEqual(incoming, existing)) {
-            configSettings.splice(configSettings.indexOf(incoming), 1);
-          }
+      if (incoming) {
+        // Remove from add list since it already exists
+        configurationSettingToAdd.splice(configurationSettingToAdd.indexOf(incoming), 1);
```

Comment:
Why do we add to `configurationSettingsToAdd` then remove ? I'm thinking if we can avoid adding all setting to `configurationSettingsToAdd` and when iterating existing keys we can determine which key values to add ?